commands/version/index.js

```diff
@@ -136,6 +137,8 @@ class VersionCommand extends Command {
       this.execOpts,
       this.options
     );
+    // only update packages that already have version field
+    this.updates = this.updates.filter(node => hasVersion(node));
```

Comment:
I would prefer we chained the filter to the `collectUpdates` call:

```js
    this.updates = collectUpdates(
      this.packageGraph.rawPackageList,
      this.packageGraph,
      this.execOpts,
      this.options
    ).filter(node => {
      if (!node.version) {
        // a package may be unversioned only if it is private
        if (node.pkg.private) {
          this.logger.info("version", "Skipping unversioned private package %j", node.name);
        } else {
          throw new ValidationError(
            "ENOVERSION",
            dedent`
              A version field is required in ${node.name}'s package.json file.
              If you wish to keep the package unversioned, it must be made private.
            `
          );
        }
      }

      return !!node.version;
    });
```

Also, instead of warning, we should definitely be throwing an error (which logs an error message) when a non-private package is missing a version field. I inlined the method because we really don't have a lot of logic to encapsulate.